tests/generator/generator.cpp

```diff
@@ -2207,6 +2640,51 @@ TEST(Generator, analyserModelScopeTest)
     EXPECT_EQ(fileContents("generator/hodgkin_huxley_squid_axon_model_1952/model.c"), generator->implementationCode());
 }
 
+TEST(Generator, daeModel)
+{
+    auto parser = libcellml::Parser::create(false);
+    auto model = parser->parseModel(fileContents("generator/dae_cellml_1_1_model/model.cellml"));
+
+    EXPECT_EQ(size_t(0), parser->errorCount());
+
+    auto analyser = libcellml::Analyser::create();
+
+    analyser->analyseModel(model);
+    printIssues(analyser);
+
+    EXPECT_EQ(size_t(0), analyser->errorCount());
+
+    auto analyserModel = analyser->model();
+
+    EXPECT_EQ(libcellml::AnalyserModel::Type::DAE, analyserModel->type());
```

Comment:
would it be useful to test the number of NLA systems (equations?) in the analysed model? (trying to remember if there is API to get that...)

Reply:
There is no API _per se_ to retrieve the number of NLA systems/equations. What we can do is to retrieve the equations associated with a model and, from there, check the type of each of those equations. If we want more then we probably ought to create a new GitHub issue?

Reply:
sounds good. We can explore future API as needed...